static/js/components/EducationForm.js

```diff
@@ -98,14 +102,59 @@ class EducationForm extends ProfileFormFields {
     </Grid>;
   };
 
-  handleSwitchClick(level){
+  entriesForLevel: Function = (level: string): number[] => {
+    const { profile } = this.props;
+    if ( _.isEmpty(profile.education) ) {
+      return [];
+    } else {
+      return Object.entries(profile.education).
+        filter(([, entry]: [string, EducationEntry]) => entry.degree_name === level).
+        map(([index,]) => Number(index));
+    }
+  };
+
+  closeDeleteAllEducationDialog: Function = (): void => {
     const {
-      ui: { educationDegreeInclusions },
+      setEducationDegreeLevel,
+      setShowEducationDeleteAllDialog,
+    } = this.props;
+    setEducationDegreeLevel('');
+    setShowEducationDeleteAllDialog(false);
+  };
+
+  deleteAllEducationEntriesForLevel: Function = (): void => {
+    const {
+      profile,
+      saveProfile,
+      ui,
+      ui: { educationDegreeLevel, educationDegreeInclusions },
       setEducationDegreeInclusions,
     } = this.props;
+    let toDelete = this.entriesForLevel(educationDegreeLevel);
+    let clone = _.cloneDeep(profile);
+    toDelete.forEach(index => clone.education.splice(index, 1));
```

Comment:
I think there's actually a bigger problem with what I have here, which is that `splice` will cause changes in which entries indexes point to....

consider:

```js
let entries = [a, b, c, d];
let toDelete = [1,2]; // entries[1] = b, entries[2] = c
toDelete.forEach(index => entries.splice(index, 1))
```

in the above in the first iteration of the loop index `1` will point to b, as we intend. After we splice that out, though, we'll have:

```js
entries = [a, c, d];
```

Now we're going to splice out the entry at index `2`, which is now `d`, and not `b`, which we had actually intended to delete. Oopsie.